algo/src/main/java/org/neo4j/graphalgo/similarity/SimilarityExporter.java

```diff
@@ -62,6 +62,29 @@ private void export(SimilarityResult similarityResult) {
 
     }
 
+    private void export(List<SimilarityResult> similarityResults) {
+        applyInTransaction(statement -> {
+            for (SimilarityResult similarityResult : similarityResults) {
+                try {
+                    createRelationship(similarityResult, statement);
+                } catch (KernelException e) {
+                    ExceptionUtil.throwKernelException(e);
```

Comment:
should we really abort the whole tx if one relationship failed?

What do you think? We could also have a counter and we only throw after a certain percentage failed?
and otherwise log them.
In algos I think we don't abort writes if a few fail b/c those are all independent tx.

Not sure though if after a single KernelException the TX is tainted/failed anyway.
Perhaps something to turn into an issue and handle later.
Also that Exception int the callback where does it land?
In `applyInTransaction` and is it logged their or thrown up?

Reply:
After we looked at the code this seems consistent with what is done elsewhere in the library, but we might look at changing that all in another commit